common/json.go

```diff
@@ -100,6 +100,11 @@ func parseJSONResponse(res *http.Response, body []byte) (*ajson.Node, error) {
 	// Unmarshall the response body into JSON
 	jsonBody, err := ajson.Unmarshal(body)
 	if err != nil {
+
+		// ajson errs when trying to unmarshal empty body, even if the req was successful so we swallow error
+		if res.StatusCode == http.StatusNoContent {
```

Comment:
This relies on SaaS APIs providing the correct HTTP status codes, which might not always be the case, a better way to present this error is to check for the length of body before trying to unmarshal it with ajson

Reply:
that's a great point. i'll lift the logic out of the json parsing helper

Reply:
I see you lifted the logic out, but I think you forgot to delete this?